python-checks/src/test/resources/checks/unusedFunctionParameter/unusedFunctionParameter.py

```diff
@@ -161,3 +161,35 @@ def param_referenced_in_comment_no_issue_3(unused_param_2):
     def param_accessed_through_pandas_no_issue(sample_df, area_of_interest): # OK
         sample_df.query('area == @area_of_interest').population
         print("hello")
+
+import abc
+class Test(abc.ABC):
+    def execute_test(self, name):
+        print("Test")
+
+class Suite(metaclass=abc.ABCMeta):
+    def execute_suite(self, name):
+        print("Suite")
+
+import abc as xyz
+class Suite2(metaclass=xyz.ABCMeta):
+    def execute_suite(self, name):
+        print("Suite")
+class Test2(xyz.ABC):
+    def execute_test(self, name):
+        print("Test")
+
+from abc import ABC as XYZ
+class Test2(XYZ):
+    def execute_test(self, name):
+        print("Test")
+
+# source :  https://stackoverflow.com/a/392767
+class FakeMetaclass(type):
```

Comment:
While it's interesting to have an actual example of a metaclass, it's not really needed to have a minimal test (since the current implementation will avoid raising issues to any class that defines a metaclass).

Since the goal of the unit tests is not necessarily to have exhaustive code, defining a metaclass such as:
`class FakeMetaClass(type): ...` would be enough. Having a test case where the metaclass is a symbol that is not even in the scope is acceptable as well, such as:
```
class ClassWithUnknownMetaClass(metaclass=UnknownMeta):
  def execute(self, name):
    print("Execute")
```